pymc_extras/statespace/models/structural/components/regression_dataclass.py

```diff
@@ -0,0 +1,539 @@
+from dataclasses import dataclass, field
+
+import numpy as np
+
+from pytensor import tensor as pt
+
+from pymc_extras.statespace.models.structural.core import Component
+from pymc_extras.statespace.utils.constants import TIME_DIM
+
+
+@dataclass
+class ParameterProperty:
+    name: str
+    shape: tuple[int, ...]
+    dims: tuple[str, ...]
+    constraints: str | None = None
+
+    def __str__(self):
+        base = f"name: {self.name}\nshape: {self.shape}\ndims: {self.dims}"
+        if self.constraints:
+            return base + f"\nconstraints: {self.constraints}"
+        return base
+
+
+@dataclass
+class ParameterProperties:
```

Comment:
```suggestion
class ParameterInfo(Info[StatespaceParameter]):
```

Same here. I'm also not against just calling it `StatespaceParameters` (plural), but that's a bit less obvious.